content/blog/tinacms-styling-system.md

```diff
@@ -0,0 +1,43 @@
+---
+title: 'TinaCMS Styling System'
+date: '2020-03-18T00:00:00.000Z'
+author: Scott Byrne
+---
+
+## Writing an adaptable UI is hard.
+
+When you're writing code for a quickly changing project, you want that code to be cheap and replaceable. [Styled Components](https://styled-components.com/) allow us to write styles right alongside the components they're used for. It lets us easily move components around — or completely replace them — with minimum effort.
+
+All these components — spread across many packages and files — need a single source of truth. Colors, padding sizes, shadows, fonts.. everything had to be consistent between TinaCMS components. Styled Components offered a powerful theming system that was a natural fit for the problem. We could provide theme context to any component looking to use common Tina styles.
+
+A major shift in our approach has been the ability to edit content directly inline where it's displayed. This meant that field specific UI would be displayed right on the user's website, not in an isolated TinaCMS container. We adapted our system to make this work, but the limitations of our approach became obvious. Even a simple button required type references to the styles package and a theme context provider. Theme changes — configured through gatsby-config.js — wouldn't take effect until the server restarted. We needed something more suited to the evolving TinaCMS project.
```

Comment:
> A major shift in our approach has been the ability to edit content
At first, I thought this was referring to our approach of using Styled Components, (and not our approach of Tina itself). Might be confusing for anyone reading this post without knowing a lot about Tina.

Reply:
I changed the first couple sentences:

> Initially, Tina UI was contained within an isolated sidebar inside an iframe. As we moved towards editing content inline with Tina, field specific UI would be displayed right on the user's website, not in an isolated TinaCMS container.